source/org/zfin/feature/repository/HibernateFeatureRepository.java

```diff
@@ -512,13 +512,37 @@ public List<FeaturePrefix> getLabPrefixes(String labName) {
         return getLabPrefixes(labName, true);
     }
 
+    /**
+     * Get the next zf line number
+     * @return
+     */
     public String getNextZFLineNum() {
-        String sql = "select max(cast(coalesce(feature_line_number,'0') as integer)) + 1 from feature where feature_lab_prefix_id = 194";
-        Query query = HibernateUtil.currentSession().createSQLQuery(sql);
-        return query.uniqueResult().toString();
+        return getNextZFLineNumWithoutFeatureTrackingCollision();
+    }
 
+    /**
+     * Get the next zf line number. First look at the feature table and add 1 to the max value.
+     * Then, check if there is a feature tracking entry with the same line number. If so, increment the line number
+     * until there is no collision.
+     * @return next line number (without collision)
+     */
+    private String getNextZFLineNumWithoutFeatureTrackingCollision() {
+        Integer nextLine = getNextZFLineNumFromFeatureTable();
+        while(getFeatureTrackingByAbbreviation("zf" + nextLine) != null) {
+            nextLine++;
+        }
+        return String.valueOf(nextLine);
     }
 
+    /**
+     * Get the next zf line number by looking at the feature table and adding 1 to the max value.
+     * @return next line number
+     */
+    private Integer getNextZFLineNumFromFeatureTable() {
+        String sql = "select max(cast(coalesce(feature_line_number,'0') as integer)) + 1 from feature where feature_lab_prefix_id = 194";
```

Comment:
I there a more generic fix? Lab-independent

Reply:
I updated the code to be lab independent. I noticed a couple of things:

1. The other labs don't use this logic to figure out the next line number. Maybe we should add that? Here's the frontend code:
```
        if (view.labDesignationBox.getSelected().equals(ZF_PREFIX)) {
            FeatureRPCService.App.getInstance().getNextZFLineNum(
...
```
2. Maybe related to #1, but the feature_line_number column of the feature table that we use for that calculation is often not an integer number like the ones for ZFIN.  Some examples: sagn025a, sagn028c, 001tk, 255H, iy121, a5, et33e20. It might be hard to come up with a generic way of finding the next "number" for those.

Reply:
Could you check with Yvonne or Holly if it should only apply to that one lab? I agree it's not possible when line numbers contain alphanumeric characters.

Reply:
I got the response in Jira:

> @Ryan Taylor no I do not think we want this for other labs as often multiple labs share a designation and have a certain range (Lab A uses numbers 1-100, Lab B uses numbers 1001-200, etc). We can get requests out of sequence and that makes it hard to manage. We do this for the zf numbers since we do not share our designation.